test-data/pom.xml

```diff
@@ -12,7 +12,7 @@
     <name>Jandex: Test Data</name>
 
     <properties>
-        <java.version>17</java.version>
+        <java.version>8</java.version>
```

Comment:
I'm not reviewing this PR properly yet, but this is not gonna work -- the `test-data` artifact uses Java 8+ features, like modules or records, so it cannot be built for Java 8.